packages/grid/x-data-grid-premium/src/tests/rowGrouping.DataGridPremium.test.tsx

```diff
@@ -2634,4 +2634,24 @@ describe('<DataGridPremium /> - Row Grouping', () => {
 
     await waitFor(() => expect(getCell(1, 3).textContent).to.equal('username 2'));
   });
+
+  // See https://github.com/mui/mui-x/issues/8580
```

Comment:
Optional Nice to have (possibly in a follow-up): It'd be nice to have a small test case covering https://github.com/mui/mui-x/issues/8853 too

Reply:
Good idea, I'll add a regression test in a follow-up PR

Reply:
Added in https://github.com/mui/mui-x/pull/8870